hadoop-ozone/dist/src/main/smoketest/snapshot/upgrade-snapshot-check.robot

```diff
@@ -0,0 +1,82 @@
+# limitations under the License.
+
+*** Settings ***
+Documentation       Smoketest ozone cluster snapshot feature
+Library             OperatingSystem
+Library             BuiltIn
+Resource            ../commonlib.robot
+Test Timeout        5 minutes
+
+*** Variables ***
+
+
+*** Test Cases ***
+Create snapshot
+    [Tags]     post-finalized-snapshot-tests
+    ${output} =         Execute                ozone sh volume create snapvolume-1
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute                ozone sh bucket create /snapvolume-1/snapbucket-1
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute                ozone sh snapshot create /snapvolume-1/snapbucket-1 snapshot1
+                        Should not contain     ${output}       Failed
+                        Execute and checkrc    echo "key created using Ozone Shell" > /tmp/sourcekey    0
+                        Execute                ozone sh key put /snapvolume-1/snapbucket-1/key1 /tmp/sourcekey
+                        Execute                ozone sh snapshot create /snapvolume-1/snapbucket-1 snapshot2
+
+Attempt to create snapshot when snapshot feature is disabled
+    [Tags]     pre-finalized-snapshot-tests
+    ${output} =         Execute And Ignore Error    ozone sh volume create snapvolume-2     
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute And Ignore Error    ozone sh bucket create /snapvolume-2/snapbucket-1     
+                        Should not contain     ${output}       Failed
+    ${output} =         Execute and checkrc         ozone sh snapshot create /snapvolume-2/snapbucket-1 snapshot1    255
+                        Should contain    ${output}   NOT_SUPPORTED_OPERATION
+
+List snapshot
+    [Tags]     post-finalized-snapshot-tests
+    ${output} =         Execute           ozone sh snapshot ls /snapvolume-1/snapbucket-1
+                        Should contain    ${output}       snapshot1
+                        Should contain    ${output}       snapshot2
+                        Should contain    ${output}       SNAPSHOT_ACTIVE
+
+Attempt to list snapshot when snapshot feature is disabled
+    [Tags]     pre-finalized-snapshot-tests
+    ${output} =         Execute and checkrc         ozone sh snapshot ls /snapvolume-2/snapbucket-1    255
+                        Should contain    ${output}   NOT_SUPPORTED_OPERATION
+
+Snapshot Diff
+    [Tags]     post-finalized-snapshot-tests
+    ${output} =         Execute           ozone sh snapshot snapshotDiff /snapvolume-1/snapbucket-1 snapshot1 snapshot2
+                        Should contain    ${output}       Snapshot diff job is IN_PROGRESS
+    ${output} =         Execute           ozone sh snapshot snapshotDiff /snapvolume-1/snapbucket-1 snapshot1 snapshot2
+                        Should contain    ${output}       +    key1
```

Comment:
Would this cause flakiness if the SnapDiff job is still in-progress and not finished at the second `snapshotDiff` CLI?

Better loop and wait until it doesn't report `IN_PROGRESS` any more?

Reply:
Agree with @smengcl, we should have loop and wait to remove any flakiness.

Reply:
yeah sure! will update!